docs/design/2020-05-08-standardize-error-codes-and-messages.md

```diff
@@ -0,0 +1,166 @@
+# Proposal: Standardize error codes and messages
+
+- Author(s):     Joshua
+- Last updated:  May 8
+- Discussion at: https://docs.google.com/document/d/1beoa5xyuToboSx6e6J02tjLLX5SWzmqvqHuAPEk-I58/edit?usp=sharing
+
+## Abstract
+
+This proposal proposes to create a document to display error codes and error messages.
+
+## Motivation
+
+When certain errors occur in the TiDB system, an error message with an error code will be thrown. 
+In many cases, users don't know what the error message is saying. So we need a manual to locate 
+errors and solutions. In order to allow users to locate the cause of the error faster, we can 
+integrate the manual to the `doc` component of [TiUP](https://github.com/pingcap-incubator/tiup),
+so that users can locate error with the command:
+
+```bash
+tiup doc desc-error <code>
+```
+
+The `code` is the error code thrown by TiDB system. Another use case is that users can search error
+with some key:
+
+```bash
+tiup doc search-error <key>
+```
+
+The `key` can be any keyword from an error message.
+
+## Proposal
+
+### Requirement
+
+In order to let TiUP know the the errors every component may throw, the components developers should
+keep a meta file in the code repository. The meta file can be a json file, a toml file or a markdown file.
+
+#### Meta file in JSON
+
+The json format of meta file is like:
+
+```json
+[
+    {
+        "code": 8005,
+        "error": "Transactions in TiDB encounter write conflicts.",
+        "message": "See [the Troubleshoot section](https://pingcap.com/docs/stable/faq/tidb#troubleshoot) for the cause and solution.",
```

Comment:
I still think we need to give a more detailed explanation here besides the troubleshooting link.

Reply:
Yes. But this is an example to show how to use markdown syntax in the message, the message content should be reviewed per PR

Reply:
an explanation is always better, it s can show the best format we recommend, otherwise, guys will write a variety of patterns 🤣

Reply:
OK